checks/folder_sftp.go

```diff
@@ -14,15 +14,24 @@ func CheckSFTP(ctx *context.Context, check v1.FolderCheck) pkg.Results {
 	result := pkg.Success(check, ctx.Canary)
 	var results pkg.Results
 	results = append(results, result)
-	auth, err := GetAuthValues(check.SFTPConnection.Auth, ctx.Kommons, ctx.Canary.Namespace)
+
+	foundConn, err := check.SFTPConnection.HydrateConnection(ctx)
 	if err != nil {
-		return results.ErrorMessage(err)
+		return results.Failf("failed to populate SFTP connection: %v", err)
```

Comment:
Why is this logic different from the others ?

Reply:
If you mean why it's using `Failf` instead of `ErrorMessage()` then it's for no particular reason. I prefer `Failf` as it adds more context.